src/builder.rs

```diff
@@ -0,0 +1,110 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use hyper::{Body, Request, Response};
+
+use crate::app_context::AppContext;
+use crate::async_graphql_hyper::{GraphQLBatchRequest, GraphQLRequest};
+use crate::blueprint::{validate_operations, Blueprint, OperationQuery};
+use crate::config::{Config, ConfigModule, ConfigReader, Source};
+use crate::fmt::Fmt;
+use crate::http::handle_request;
+use crate::print_schema;
+use crate::runtime::TargetRuntime;
+use crate::valid::Validator;
+
+#[derive(Clone)]
+pub struct TailcallBuilder {
+    files: Vec<String>,
+    schemas: Vec<SchemaHolder>,
+    runtime: TargetRuntime,
+}
+#[derive(Clone)]
+struct SchemaHolder {
+    source: Source,
+    schema: String,
+    parent: Option<String>,
+}
+
+#[derive(Clone)]
+pub struct TailcallExecutor {
+    pub config_module: ConfigModule,
+    pub app_ctx: Arc<AppContext>,
+}
+
+impl TailcallBuilder {
+    pub fn init(runtime: TargetRuntime) -> Self {
```

Comment:
```suggestion
    pub fn new() -> Self {
```

Pass the runtime in the end using the `build` method.